src/wallet/wallet.cpp

```diff
@@ -4022,21 +4030,22 @@ int CWallet::ScanForWalletTransactions(
             // Only attempt to perform scans for Orchard during wallet initialization,
             // since we do not support Orchard key import.
             if (isInitScan) {
```

Comment:
```suggestion
            if (isInitScan && pindex->nHeight >= 1) {
```

Reply:
`pindex->nHeight` is an int, and so taking `max(nu5_height.value, pindex->nHeight - 1)` can validly return 0 (and we would want to rewind to that height) if Orchard were active from genesis.